docs/documentation/upgrading/topics/keycloak/changes-24_0_0.adoc

```diff
@@ -174,6 +174,25 @@ The `--proxy` option has been deprecated and will be removed in a future release
 NOTE: For hardened security, the `--proxy-headers` option does not allow selecting both `forwarded` and `xforwarded` values at the same time (as it was
 the case before for `--proxy edge` and `--proxy reencrypt`).
 
+WARNING: When using the proxy headers option, make sure your reverse proxy properly sets and overwrites the `Forwarded` or `X-Forwarded-*` headers respectively.
```

Comment:
Do we know that overwriting is required? How does the forwarder parser handle things if there are multiple values?

Reply:
Looking at Quarkus' [impl](https://github.com/quarkusio/quarkus/blob/5044d93f0e139342042ec251e2e2ac4eabd0a804/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java#L128), it accepts the first header found. But I think it's undefined behaviour, we can't rely on the order. That's why we strongly suggest overwrtiting the headers.